models/car.go

```diff
@@ -0,0 +1,13 @@
+package models
+
+import "gorm.io/gorm"
+
+type Car struct {
+	gorm.Model
+	LicensePlate string    `json:"licensePlate"`
+	Brand        string    `json:"brand"`
+	CarModel     string    `json:"model"`
+	Color        string    `json:"color"`
+	PlateNumber  string    `json:"plateNumber"`
+	Parking      []Parking `json:"parking,omitempty"`
```

Comment:
why are we using parking in car model?

Reply:
Using the parking slice in the car model allows us to retrieve all parking records for a specific car easily,this can be useful for generating reports and calculating revenue

Reply:
remove this , we dont need it here , your logic will get records based on that